### DISCUSSION TITLE
How to update only one element in the scene

### DISCUSSION THREAD

**Discussion by essadkim** [excalidraw/excalidraw#8608] [Q&A]
Does the updateScene API re-render the entire scene when called like this?

`api.updateScene({elements: element})`

If so, is there a way to update only a subset of elements without re-rendering the entire scene? I believe an API that allows for partial updates would be beneficial, especially for use cases like animations where performance optimization is crucial.

**Comment by Mrazator** [comment]
ATM you will have to pass in the whole elements array, but it does not mean all elements will re-render, as their offscreen canvases get cached based on their `version`. That said, in the future we will likely have an incremental API.
